plugin/pkg/admission/podnodeselector/admission_test.go

```diff
@@ -156,47 +157,54 @@ func TestPodAdmission(t *testing.T) {
 			testName:                        "Default node selector conflict with the whitelist",
 		},
 	}
-	for _, test := range tests {
-		if !test.ignoreTestNamespaceNodeSelector {
-			namespace.ObjectMeta.Annotations = map[string]string{"scheduler.alpha.kubernetes.io/node-selector": test.namespaceNodeSelector}
-			informerFactory.Core().InternalVersion().Namespaces().Informer().GetStore().Update(namespace)
-		}
-		handler.clusterNodeSelectors = make(map[string]string)
-		handler.clusterNodeSelectors["clusterDefaultNodeSelector"] = test.defaultNodeSelector
-		handler.clusterNodeSelectors[namespace.Name] = test.whitelist
-		pod.Spec = api.PodSpec{NodeSelector: test.podNodeSelector}
 
-		err := handler.Admit(admission.NewAttributesRecord(pod, nil, api.Kind("Pod").WithVersion("version"), "testNamespace", namespace.ObjectMeta.Name, api.Resource("pods").WithVersion("version"), "", admission.Create, nil))
-		if test.admit && err != nil {
-			t.Errorf("Test: %s, expected no error but got: %s", test.testName, err)
-		} else if !test.admit && err == nil {
-			t.Errorf("Test: %s, expected an error", test.testName)
-		}
-		if test.admit && !labels.Equals(test.mergedNodeSelector, labels.Set(pod.Spec.NodeSelector)) {
-			t.Errorf("Test: %s, expected: %s but got: %s", test.testName, test.mergedNodeSelector, pod.Spec.NodeSelector)
-		}
-		err = handler.Validate(admission.NewAttributesRecord(pod, nil, api.Kind("Pod").WithVersion("version"), "testNamespace", namespace.ObjectMeta.Name, api.Resource("pods").WithVersion("version"), "", admission.Create, nil))
-		if test.admit && err != nil {
-			t.Errorf("Test: %s, expected no error but got: %s", test.testName, err)
-		} else if !test.admit && err == nil {
-			t.Errorf("Test: %s, expected an error", test.testName)
-		}
+	podNodeSelectorAnnotations := []string{algorithm.AnnotationNamespaceNodeSelector, algorithm.DeprecatedAnnotationNamespaceNodeSelector}
+	for i := 0; i < len(podNodeSelectorAnnotations); i++ {
```

Comment:
rather than nesting loops, make the deprecated annotation one of the inputs in the testcase, and test combinations of old and new annotations set and the resulting selector:

old annotation: "a=1"
new annotation: "b=2"
result: {a:1, b:2}

old annotation: "a=1"
new annotation: "a=2"
result: error

Reply:
Is that the logic you want? The old and new annotation merged? I'm not sure I see the point in that?